fe/fe-core/src/main/java/org/apache/doris/transaction/DatabaseTransactionMgr.java

```diff
@@ -1436,9 +1501,8 @@ public void unprotectWriteAllTransactionStates(DataOutput out) throws IOExceptio
             entry.getValue().write(out);
         }
 
-        for (TransactionState transactionState : finalStatusTransactionStateDeque) {
+        for (TransactionState transactionState : idToFinalStatusTransactionState.values()) {
```

Comment:
use two finalStatusTransactionStateDeques instead of idToFinalStatusTransactionState to keep order for transaction state

Reply:
Done